src/dune_engine/cached_digest.ml

```diff
@@ -332,7 +332,7 @@ let remove path =
   Path.Table.remove cache.table path
 
 module Untracked = struct
-  let source_or_external_file = peek_or_refresh_file ~allow_dirs:false
+  let source_or_external_file = peek_or_refresh_file ~allow_dirs:true
```

Comment:
I'm a little confused by this change. Why is it needed?

Reply:
It's not needed. It was a leftover from earlier iterations. I updated the PR and this is gone.